### DISCUSSION TITLE
setup.sh: Exit status 14 with BIO_should_retry at installation

### DISCUSSION THREAD

**Discussion by LPe7** [winapps-org/winapps#865] [Support]
Hi,

I'm trying to install winapps with absolutely *no containerization knowledge* at all. I'm following both winapps' guide and [this gist](https://gist.github.com/eylenburg/38e5da371b7fedc0662198efc66be57b) to install office 365.
I'm using podman on ubuntu 24.04 (with gnome wayland), all the dependencies are installed.

`~/.config/winapps/winapps.conf`: (password and username correspond to what is set in windows in the vm)

```
RDP_USER="Docker"
RDP_PASS="password"
WAFLAVOR="podman"
DEBUG="true"
```

`./setup.sh`'s output:

```
[...]
Attempting to establish a Remote Desktop connection with Windows... Failed!

ERROR: REMOTE DESKTOP PROTOCOL FAILURE.
FreeRDP failed to establish a connection with Windows.
--------------------------------------------------------------------------------
Please view the log at /home/elio/.local/share/winapps/FreeRDP_Test_20241026_2308_854749302.log.
Troubleshooting Tips:
[...]
--------------------------------------------------------------------------------
Exiting with status '14'.
```

`/home/user/.local/share/winapps/FreeRDP_Test_20241026_2151_382847899.log`:

```
[21:51:13:510] [99275:000183cc] [WARN][com.freerdp.core.nego] - [nego_enable_aad]: This build does not support AAD security, disabling.
[21:51:16:579] [99275:000183cc] [ERROR][com.freerdp.core.transport] - [transport_default_write]: BIO_should_retry returned a system error 32: Relais brisé (pipe)
[21:51:16:579] [99275:000183cc] [ERROR][com.freerdp.core] - [transport_default_write]: ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]
```

I [read](https://github.com/winapps-org/winapps/issues/76#issuecomment-2014706467) that it should be a networking issue and that i should check my vm's ip address and its connection, but i absolutely don't know how to do it :laughing: ...
I ran `podman container inspect -f '{{.NetworkSettings.IPAddress}}' WinApps` but it does'nt return anything.

Also, I don't get any new file in `~/.config/freerdp/server`.

Could anyone help the newbie I am please ? thank you very very much !

**Comment by SergioLoaiza** [comment]
Hi, I got the same error; any hint of how to resolve i?

**Comment by mattcbro** [comment]
I also am having a lot of trouble with the installation due to a failure to connect to the windows virtual machine through RDP.  I have been able to get xfreerdp to connect to the virtual machine directly from the command line.

Your virtual machines IP address should be 127.0.0.1 unless you created one earlier yourself.   Once your VM has been created via, 
```docker compose --file ~/.config/winapps/compose.yaml up```
You should be able to use the remote desktop protocol software to login in to your vm.   I can log in to mine using
```xfreerdp /u:matt /v:127.0.0.1```
In the vm I created the user matt with a password.  I entered that user and password into my winapps.conf file under
```RDP_USER="matt"```
and the password
```RDP_PASS="my_private_password"```

This allows the install to get further.  I actually see the vm window come up,  but then it hangs trying to log in.  I have no idea why.

Another super annoying thing is that once the setup has failed,  you keep having to remove the directory
```rm -rf ~/.local/bin/winapps-src/```
to get it to run again.

**Comment by progzone122** [comment]
I had the same problem when I tried to connect via RDP using the command specified in the docs.

```shell
➜  ~ xfreerdp3 /u:"user" /p:"user" /v:127.0.0.1 /cert:tofu /sec:rdp
...
[12:02:16:220] [26432:00006742] [ERROR][com.freerdp.core] - [transport_read_layer]: ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]
[12:02:16:248] [26432:00006742] [ERROR][com.freerdp.core.transport] - [transport_read_layer]: BIO_read retries exceeded
[12:02:16:248] [26432:00006742] [ERROR][com.freerdp.core] - [transport_read_layer]: ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]
[12:02:16:248] [26432:00006742] [ERROR][com.freerdp.core] - [freerdp_connect]: freerdp_post_connect failed
```

It helped me to **completely disable the firewall and omit the `/cert:tofu` and `/sec:rdp` arguments when connecting via xfreerdp3.**

_P.S. It’s not necessary to completely disable the firewall, but you can do it for testing purposes, and then turn it back on and set the correct rules._

```shell
xfreerdp3 /u:"user" /p:"user" /v:127.0.0.1
```

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/25327f4f-67f0-4ee8-805c-6f3167d8313a" />

**Comment by progzone122** [comment]
> Another super annoying thing is that once the setup has failed, you keep having to remove the directory
> `rm -rf ~/.local/bin/winapps-src/`

No need to do that! Just delete volume using docker
```shell
sudo docker volume rm winapps_data
```
